lib/txmgr/txmgr.go

```diff
@@ -677,3 +686,14 @@ func maybeSetTimeout(ctx context.Context, timeout time.Duration) (context.Contex
 
 	return context.WithTimeout(ctx, timeout)
 }
+
+// isNetworkError returns true if the error is a network error.
+func isNetworkError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	opErr := new(net.OpError)
+
+	return errors.As(err, &opErr)
```

Comment:
Any reason we're using `As` instead of `Is` here?

Reply:
Because it doesn't work.

```
func TestScratch(t *testing.T) {
	e1 := error(&net.OpError{Op: "dial", Net: "tcp", Source: nil, Addr: nil, Err: io.EOF})
	e2 := fmt.Errorf("wrapped: %w", e1)
	e3 := errors.Wrap(e2, "wrapped2")

	zeroOpErr := new(net.OpError)

	fmt.Printf("🔥!! =%v\n", errors.Is(e3, e1))
	fmt.Printf("🔥!! =%v\n", errors.Is(e3, e2))
	fmt.Printf("🔥!! =%v\n", errors.Is(e3, e3))
	fmt.Printf("🔥!! =%v\n", errors.Is(e3, zeroOpErr))
	fmt.Printf("🔥!! =%v\n", errors.As(e3, &zeroOpErr))
}
```


Results in:
```
🔥!! =true
🔥!! =true
🔥!! =true
🔥!! =false
🔥!! =true

```

Reply:
TL;DR:
 - `As`: check for types of errors
 - `Is`: checks for instances of errors.

Reply:
thanks for the explanation, i think i finally understand this :D